packages/calcite-components/src/components/table-row/table-row.tsx

```diff
@@ -335,71 +348,58 @@ export class TableRow extends LitElement implements InteractiveComponent {
             ? "circle-f"
             : "circle";
 
-    return <calcite-icon icon={icon} scale={getIconScale(this.scale)} />;
+    icon.icon = iconName;
+    icon.scale = getIconScale(this.scale);
+
+    return icon;
   }
 
-  private renderSelectableCell(): JsxNode {
-    return this.rowType === "head" ? (
-      <calcite-table-header
-        alignment="center"
-        bodyRowCount={this.bodyRowCount}
-        key="selection-head"
-        onClick={this.selectionMode === "multiple" && this.handleSelectionOfRow}
-        onKeyDown={this.selectionMode === "multiple" && this.handleKeyboardSelection}
-        parentRowAlignment={this.alignment}
-        selectedRowCount={this.selectedRowCount}
-        selectedRowCountLocalized={this.selectedRowCountLocalized}
-        selectionCell={true}
-        selectionMode={this.selectionMode}
-      />
-    ) : this.rowType === "body" ? (
-      <calcite-table-cell
-        alignment="center"
-        key="selection-body"
-        onClick={this.handleSelectionOfRow}
-        onKeyDown={this.handleKeyboardSelection}
-        parentRowAlignment={this.alignment}
-        parentRowIsSelected={this.selected}
-        parentRowPositionLocalized={this.positionSectionLocalized}
-        selectionCell={true}
-      >
-        {this.renderSelectionIcon()}
-      </calcite-table-cell>
-    ) : (
-      <calcite-table-cell
-        alignment="center"
-        key="selection-foot"
-        parentRowAlignment={this.alignment}
-        selectionCell={true}
-      />
-    );
+  private renderNumberedCell(): void {
+    const cell =
+      this.rowType === "head"
+        ? document.createElement("calcite-table-header")
+        : document.createElement("calcite-table-cell");
+
+    cell.alignment = "center";
+    cell.numberCell = true;
+    cell.parentRowAlignment = this.alignment;
+
+    if (this.rowType === "body") {
+      cell.textContent = this.positionSectionLocalized;
+    }
+
+    this.tableRowEl.append(cell);
   }
 
-  private renderNumberedCell(): JsxNode {
-    return this.rowType === "head" ? (
-      <calcite-table-header
-        alignment="center"
-        key="numbered-head"
-        numberCell={true}
-        parentRowAlignment={this.alignment}
-      />
-    ) : this.rowType === "body" ? (
-      <calcite-table-cell
-        alignment="center"
-        key="numbered-body"
-        numberCell={true}
-        parentRowAlignment={this.alignment}
-      >
-        {this.positionSectionLocalized}
-      </calcite-table-cell>
-    ) : (
-      <calcite-table-cell
-        alignment="center"
-        key="numbered-foot"
-        numberCell={true}
-        parentRowAlignment={this.alignment}
-      />
-    );
+  private renderSelectableCell(): void {
+    const cell: any =
```

Comment:
(optional) store the reference to the element that was created on previous render and update it rather than create a new element?
that would improve re-render performance